### DISCUSSION TITLE
How can I hook an application that uses split apks? (Unsupported class loader)

### DISCUSSION THREAD

**Discussion by elesbb** [LSPosed/LSPosed#2483] [Q&A] [answered]
I have a textra mod that enables some additional features I made (close on send, show keyboard, etc)

It was working great until the recent update. Textra will immediately crash.

Here is the logcat for textra when I try to launch it:

    04-11 00:42:52.830 30933 30933 W com.textra: Unsupported class loader
    04-11 00:42:52.831 30933 30933 I LSPosed-Bridge:   Loading class com.elesbb.pixel7xposed.XposedMain
    04-11 00:42:52.840 30933 30933 D CompatibilityChangeReporter: Compat change id reported: 171979766; UID 10283; state: ENABLED
    04-11 00:42:52.846 30933 30933 D nativeloader: Configuring classloader-namespace for other apk /system/framework/org.apache.http.legacy.jar. target_sdk_version=31, uses_libraries=ALL, library_path=/data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/lib/arm64:/data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/base.apk!/lib/arm64-v8a:/data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/split_config.arm64_v8a.apk!/lib/arm64-v8a:/data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/split_config.en.apk!/lib/arm64-v8a:/data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/split_config.xxxhdpi.apk!/lib/arm64-v8a, permitted_path=/data:/mnt/expand:/data/user/0/com.textra
    04-11 00:42:52.851 30933 30933 W ziparchive: Unable to open '/data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/split_config.arm64_v8a.dm': No such file or directory
    04-11 00:42:52.851 30933 30933 W ziparchive: Unable to open '/data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/split_config.arm64_v8a.dm': No such file or directory
    04-11 00:42:52.851 30933 30933 W com.textra: Entry not found
    04-11 00:42:52.851 30933 30933 W ziparchive: Unable to open '/data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/split_config.en.dm': No such file or directory
    04-11 00:42:52.851 30933 30933 W ziparchive: Unable to open '/data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/split_config.en.dm': No such file or directory
    04-11 00:42:52.851 30933 30933 W com.textra: Entry not found
    04-11 00:42:52.851 30933 30933 W ziparchive: Unable to open '/data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/split_config.xxxhdpi.dm': No such file or directory
    04-11 00:42:52.851 30933 30933 W ziparchive: Unable to open '/data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/split_config.xxxhdpi.dm': No such file or directory
    04-11 00:42:52.851 30933 30933 W com.textra: Entry not found
    04-11 00:42:52.852 30933 30933 D nativeloader: Configuring classloader-namespace for other apk /data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/base.apk:/data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/split_config.arm64_v8a.apk:/data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/split_config.en.apk:/data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/split_config.xxxhdpi.apk. target_sdk_version=31, uses_libraries=, library_path=/data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/lib/arm64:/data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/base.apk!/lib/arm64-v8a:/data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/split_config.arm64_v8a.apk!/lib/arm64-v8a:/data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/split_config.en.apk!/lib/arm64-v8a:/data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/split_config.xxxhdpi.apk!/lib/arm64-v8a, permitted_path=/data:/mnt/expan
    --------- beginning of crash
    04-11 00:42:52.872 30933 30933 F libc    : Fatal signal 11 (SIGSEGV), code 2 (SEGV_ACCERR), fault addr 0xb4000073bd61c000 in tid 30933 (com.textra), pid 30933 (com.textra)
    04-11 00:42:52.899 30958 30958 I crash_dump64: obtaining output fd from tombstoned, type: kDebuggerdTombstoneProto
    04-11 00:42:52.900   680   680 I tombstoned: received crash request for pid 30933
    04-11 00:42:52.900 30958 30958 I crash_dump64: performing dump of process 30933 (target tid = 30933)
    04-11 00:42:53.077 30958 30958 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
    04-11 00:42:53.077 30958 30958 F DEBUG   : Build fingerprint: 'google/cheetah/cheetah:13/TQ2A.230305.008.C1/9619669:user/release-keys'
    04-11 00:42:53.077 30958 30958 F DEBUG   : Revision: 'MP1.0'
    04-11 00:42:53.077 30958 30958 F DEBUG   : ABI: 'arm64'
    04-11 00:42:53.077 30958 30958 F DEBUG   : Timestamp: 2023-04-11 00:42:52.904368070-0400
    04-11 00:42:53.077 30958 30958 F DEBUG   : Process uptime: 1s
    04-11 00:42:53.077 30958 30958 F DEBUG   : Cmdline: com.textra
    04-11 00:42:53.077 30958 30958 F DEBUG   : pid: 30933, tid: 30933, name: com.textra  >>> com.textra <<<
    04-11 00:42:53.077 30958 30958 F DEBUG   : uid: 10283
    04-11 00:42:53.077 30958 30958 F DEBUG   : tagged_addr_ctrl: 0000000000000001 (PR_TAGGED_ADDR_ENABLE)
    04-11 00:42:53.077 30958 30958 F DEBUG   : signal 11 (SIGSEGV), code 2 (SEGV_ACCERR), fault addr 0xb4000073bd61c000
    04-11 00:42:53.077 30958 30958 F DEBUG   :     x0  0000000000000001  x1  0000000000000002  x2  000000000000e4f9  x3  000000000000e4fd
    04-11 00:42:53.077 30958 30958 F DEBUG   :     x4  000000000000e501  x5  000000002e8da2f0  x6  00000000bfa2b0ed  x7  7f7fff7fffffff7f
    04-11 00:42:53.077 30958 30958 F DEBUG   :     x8  b4000073bd600570  x9  b4000073bd606a75  x10 000000000001558b  x11 00000100000001b3
    04-11 00:42:53.077 30958 30958 F DEBUG   :     x12 0000000000000000  x13 74236763c5b5a21c  x14 00000000a5a389ee  x15 00000000001aace3
    04-11 00:42:53.077 30958 30958 F DEBUG   :     x16 000000004b4713dc  x17 00000000000d5671  x18 00000073c79c6000  x19 0000007ff06f9ff0
    04-11 00:42:53.077 30958 30958 F DEBUG   :     x20 000000000001ba8b  x21 60faef09a12738ba  x22 00000000597f683e  x23 00000000d94cdd18
    04-11 00:42:53.077 30958 30958 F DEBUG   :     x24 00000000ffffd1b3  x25 0000000000009c2e  x26 74236763c5b5a21c  x27 0000000001be8d95
    04-11 00:42:53.077 30958 30958 F DEBUG   :     x28 b4000073bd600570  x29 000000000001a93e
    04-11 00:42:53.077 30958 30958 F DEBUG   :     lr  00000070f50d3d94  sp  0000007ff06f9e20  pc  00000070f50d3dc0  pst 0000000000001000
    04-11 00:42:53.077 30958 30958 F DEBUG   : backtrace:
    04-11 00:42:53.077 30958 30958 F DEBUG   :       #00 pc 0000000000037dc0  /data/app/~~xfQIunMKmCzbPwU7qidwAw==/com.textra-e-66LFqKc1l7S1F1vUdfcA==/split_config.arm64_v8a.apk!libpairipcore.so

Or is this an anti xposed approach? Something they implemented to prevent xposed from hooking their apps?

**Comment by yujincheng08** [comment, marked as answer]
this an anti xposed

**Reply by elesbb** [reply]
That's annoying. Do you know any way around it? It appears like they are using a lib file (.so) so it may be difficult/not worth it. I may have to just stick to an older version or make the edits in smali code
